File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> list=new ArrayList<>();
        backtrack(nums,list,new ArrayList<>(),0);
        return list;
    }
    void backtrack(int nums[],List<List<Integer>> list, List<Integer> curr,int index){
        if(index==nums.length){
            list.add(new ArrayList<>(curr));
            return;
        }
        curr.add(nums[index]);
        backtrack(nums,list,curr,index+1);
        curr.remove(curr.size()-1);
        backtrack(nums,list,curr,index+1);
    }
}

File: 0078-subsets/0078-subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        res=[]
        def bt(index , curr):
            res.append(curr[:])
            for i in range(index,len(nums)):
                curr.append(nums[i])
                bt(i+1,curr)
                curr.pop()
        bt(0,[])
        return res        